portal-api/api-tests/v1/gef/application.api-test.js

```diff
@@ -366,6 +368,39 @@ describe(baseUrl, () => {
         expect(getFacilityResponse.body.details.submittedAsIssuedDate).toEqual(expect.any(String));
       });
 
+      it('adds coverStartDate if cover starts on submission (shouldCoverStartOnSubmission === true)', async () => {
+        // create deal
+        const { body } = await as(aMaker).post(mockApplications[0]).to(baseUrl);
+        const applicationId = body._id;
+        // create issued facility that's associated with the deal
+        const issuedFacility = mockFacilities.find((f) => f.shouldCoverStartOnSubmission === true);
+
+        const createFacilityResponse = await as(aMaker).post({ applicationId, ...issuedFacility }).to(facilitiesUrl);
+        expect(createFacilityResponse.status).toEqual(201);
+
+        const facilityId = createFacilityResponse.body.details._id;
+        // check that the facility does not already have submittedAsIssuedDate
+        expect(createFacilityResponse.body.details.submittedAsIssuedDate).toEqual(null);
```

Comment:
Can we add the same check for `shouldCoverStartOnSubmission`

Reply:
done